charts/tezos/values.yaml

```
@@ -243,6 +243,10 @@ expected_proof_of_work: 26
 node_config_network:
   chain_name: mainnet
 
+# If you dont want to join public network but need to set networks name to a
+# public one
```

```suggestion
# Setting node_config_network with a chain_name assumes the network is public.
# tezos-k8s will then configure the nodes to connect to boot-nodes for the selected chain.
# This may not always be desired. For example, you may want to run a mainnet replica
# in isolation with yes-nodes.
# Uncomment the line below to remove boot-node configuration from nodes on a public
# network. The nodes will not establish any connection outside of the cluster. 
```

---

@nicolasochem Perhaps the new flag should moved under `node_config_network` to keep the context of these fields together?

When defining a private chain, the format isn't exactly as a node expects a config.json to be because of the `activation_account_name`. I think it would be ok to put the new flag under `node_config_network` too. In addition to your comment, it should be made clear that `genesis` and  `join_public_network` are mutually exclusive, and there should be an error check in Helm to fail hard and fast if the conditions contradict. Also see comment in config-generator.py in regards to doing this validation in the backend.

I think an appropriate place for this validation is here, and it should fail if both `genesis` and `join_public_network` are true. In fact this check should return false if `!genesis and !join_public_network`. config-init container doesn't need to run if you want to run a public network node without actually joining the chain: https://github.com/oxheadalpha/tezos-k8s/blob/293abe0de4eaa3bf9c405b88e2ee95fecbf3643f/charts/tezos/templates/_helpers.tpl#L82-L94

---

Actually after more investigation, It seems we should still run config-init, as a url to a network configuration can be passed as the `chain_name` in values.yaml. We need config-init to fetch the json blob to produce the config.json. If the value passed was simply just the name of the network, then we wouldn't need config-init to run. We could just get the network name directly from `chain_name`.